src/TrackProgress/SFA.DAS.TrackProgress/Application/Commands/TrackProgressCommand.cs

```diff
@@ -56,8 +57,14 @@ public async Task<TrackProgressResponse> Handle(TrackProgressCommand request, Ca
                 return new TrackProgressResponse(HttpStatusCode.NotFound, "Apprenticeship not found");
         }
 
-        if (apprenticeshipResult.Body.Apprenticeships?.Count(x => x.StartDate == request.PlannedStartDate) > 1)
-            return new TrackProgressResponse(HttpStatusCode.NotFound, "Multiple results for start date");
+        if (apprenticeshipResult.Body.TotalApprenticeshipsFound > 1)
+            apprenticeshipResult.Body.Apprenticeships?.RemoveAll(x => x.StartDate == x.StopDate);
+
+        if (apprenticeshipResult.Body.Apprenticeships?.Count == 0)
+            return new TrackProgressResponse(HttpStatusCode.NotFound, "Apprenticeship not found");
+
+        if (apprenticeshipResult.Body.Apprenticeships?.Count > 1)
+            return new TrackProgressResponse(HttpStatusCode.BadRequest, "Multiple apprenticeship records exist");
```

Comment:
I think the idea of the error conditions was to return an Error object, which would be different from the Success Object. 

The error object will contain a list of errors (possibly, field, code and description). This is because we will very soon be returning multiple errors. however I'll let Ben make the call on this as for now we could get away with this pattern   

A nice example of returning different types from a controller action is here

https://github.com/SkillsFundingAgency/das-apim-endpoints/blob/master/src/VacanciesManage/SFA.DAS.VacanciesManage.Api/Controllers/VacancyController.cs#L37-L40

Reply:
Understood, now returns ProblemDetails which satisfies the Jira